app/templating/view_context.py

```diff
@@ -86,17 +87,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
     return context
 
 
-def build_view_context_for_summary(metadata, answer_store, schema_context, variables, csrf_token):
-    rendered_schema_json = renderer.render(g.schema.json, **schema_context)
-    summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json['sections'],
-                                                               answer_store, metadata)
+def build_view_context_for_summary(metadata, answer_store, schema_context, block_type, variables, csrf_token,
+                                   current_location):
+
+    summary_rendered_context = None
+
+    if block_type == 'Summary':
+        sections = renderer.render(g.schema.json, **schema_context)['sections']
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
```

Comment:
A little petty but it's clearer to show that the `build_summary_rendering_context` is doing the same thing for both summary pages if we moved it out of the ifs and pass a common variable into it. Maybe `section_list`. The section_list would just be a single item for the SectionSummary.
It might also be worth renaming `summary_rendered_context` to `summary_rendering_context` as it's the context used for rendering the summary. Yes it took me a while to get my head around that.